pkg/agent/hostagent.go

```diff
@@ -342,22 +347,43 @@ func (c *HostAgent) GetUpdatedYAMLPath() (string, error) {
 
 	return c.OtelConfigFile, nil
 }
+func (c *HostAgent) isIPPortFormat(endpoint string) bool {
+	//Regex for IPv4:PORT format, also consider localhost as an IP
+	regexPattern := `(?:localhost|((?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))):(?:[0-9]|[1-9][0-9]{1,4}|[1-5][0-9]{4}|6[0-5][0-5][0-3][0-5])$`
 
-func (c *HostAgent) checkIntConfigValidity(integrationType IntegrationType, configPath string) bool {
-	if configPath != "" {
+	// Compile the regular expression
+	regex, err := regexp.Compile(regexPattern)
+	if err != nil {
+		c.logger.Error("error invalid regexPattern for IP:Port format", zap.Error(err))
+		return false
+	}
+
+	// Path variable is IP:PORT for these integrations -> Clickhouse, Redpanda
+	if regex.MatchString(endpoint) {
+		c.logger.Info("found valid endpoint", zap.String("endpoint", endpoint))
+		return true
+	}
+	return false
+}
+func (c *HostAgent) checkIntConfigValidity(integrationType IntegrationType, cnf integrationConfiguration) bool {
```

Comment:
```suggestion
}

func (c *HostAgent) checkIntConfigValidity(integrationType IntegrationType, cnf integrationConfiguration) bool {
```
Leave empty line between two functions for more readability.